internal/config/config.go

```diff
@@ -0,0 +1,198 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+// validateDistributions checks that the condition that
+// two distributions with the same distribution ID MUST NOT share the same prefix.
+// or in other terms, every pair of id,prefix (Distribution) must be unique
+func validateDistributions(distributions distributionsMap) error {
+	uniqueMap := make(map[string]struct{})
+
+	for _, value := range distributions {
+		hash := value.hashKey()
+		if _, ok := uniqueMap[hash]; ok {
+			return fmt.Errorf("error parsing configuration: distribution value duplicated id: %s prefix: %s", value.ID, value.Prefix)
+		}
+
+		uniqueMap[hash] = struct{}{}
+	}
+
+	return nil
+}
+
+func validateEntitlements(entitlements entitlementsMap, distributions distributionsMap) error {
+	entitlementDistros := make(map[string]string)
+
+	// build lookup map, used also to deduplicate distribution names
+	for eName, distros := range entitlements {
+		for _, distro := range distros {
+			entitlementDistros[distro] = eName
```

Comment:
nit: instead of having L40 to L44 below, can simply check for existence of distro in distributions here...

```suggestion
			if _, ok := distributions[distro]; !ok {
			    return fmt.Errorf("error parsing configuration: distribution %s in entitlement %s is not configured", dName, eName)
			}
```

Then no need for entitlementDistros

Reply:
👍